test/e2e/lazy_activation_policy_test.go

```diff
@@ -90,6 +114,64 @@ var _ = ginkgo.Describe("Lazy activation policy testing", func() {
 			})
 		})
 
+		// Simple Case 3 (Lazy to immediate)
+		// refer: https://github.com/karmada-io/karmada/blob/master/docs/proposals/scheduling/activation-preference/lazy-activation-preference.md#simple-case-3-lazy-to-immediate
+		ginkgo.It("Simple Case 3 (Lazy to immediate)", func() {
+			ginkgo.By("step 1: deployment propagate success when policy created before it", func() {
+				waitDeploymentPresentOnCluster(originalCluster, namespace, deploymentName)
+			})
+
+			ginkgo.By("step 2: after policy updated (cluster=member2, remove lazy activationPreference field), deployment's propagation status changes", func() {
+				// 1. update policy placement with clusterAffinities
+				// 2. remove lazy activationPreference field
+				policy.Spec.Placement.ClusterAffinity = nil
+				policy.Spec.Placement.ClusterAffinities = []policyv1alpha1.ClusterAffinityTerm{{
+					AffinityName:    "group1",
+					ClusterAffinity: policyv1alpha1.ClusterAffinity{ClusterNames: []string{modifiedCluster}},
+				}}
+				policy.Spec.ActivationPreference = ""
+				framework.UpdatePropagationPolicyWithSpec(karmadaClient, policy.Namespace, policy.Name, policy.Spec)
+				// wait to distinguish whether the state will not change or have no time to change
+				time.Sleep(waitIntervalForLazyPolicyTest)
```

Comment:
since deployment propagation status will change, here is no need to call `Sleep`